src/validate/validate_state_property.ts

```diff
@@ -0,0 +1,214 @@
+import {ValidationError} from '../error/validation_error';
+import {getType} from '../util/get_type';
+import {deepUnbundle, unbundle} from '../util/unbundle_jsonlint';
+import {isObjectLiteral} from '../util/is_object_literal';
+
+interface ValidateStatePropertyOptions {
+    key: string;
+    value: unknown;
+}
+
+export function validateStateProperty({
```

Comment:
Is it possible that this method will return a list of validation errors? I see it only returns an array of a single error, am I missing anything?

Reply:
Array validator can return multiple errors, others return array for consistency.

I've added a test for that path:
![image](https://github.com/user-attachments/assets/1a25b14d-e6f6-4b1f-a5f0-4b7b32d6f55e)